src/branchandbound.jl

```diff
@@ -1,101 +1,59 @@
 # univariate
-function enclose_BranchandBound(f::Function, dom::Interval; order=10, tol=0.6)
-    x0 = Interval(mid(dom))
-    x = TaylorModel1(order, x0, dom)
-    return branchandbound(f(x-x0).pol, dom, tol)
-end
 
-# multivariate
-function enclose_BranchandBound(f::Function, dom::IntervalBox{D};
-                                order=10, tol=0.6) where {D}
-    x0 = mid(dom)
-    set_variables(Float64, "x", order=2order, numvars=D)
-    x = [TaylorModelN(i, order, IntervalBox(x0), dom) for i=1:D]
-    return branchandbound(f(x...).pol, dom - x0, tol)
+@inline function enclose(f::Function, X::Interval, ba::BranchAndBoundEnclosure;
+                         df=t->ForwardDiff.derivative(f, t))
+    return _branch_bound(ba, f, X, df)
 end
 
-@inline _Rnext(R::Vector{<:Interval}) = Interval(minimum(inf.(R)), maximum(sup.(R)))
+@inline function enclose(f::Function, X::IntervalBox, ba::BranchAndBoundEnclosure;
+                         df=t->ForwardDiff.gradient(w->f(w...), t.v))
+    return _branch_bound(ba, f, X, df)
+end
 
-const Kmax = 1000
+function _branch_bound(ba::BranchAndBoundEnclosure, f::Function, X::Interval_or_IntervalBox, df;
+                       initial=emptyinterval(first(X)),
+                       cnt=1)
 
-function branchandbound(p::Union{Taylor1, TaylorN},
-                        dom::Interval_or_IntervalBox,
-                        tol::Number)
-    K = 1
-    Rprev = evaluate(p, dom)
-    D = bisect(dom)
-    R = [evaluate(p, D[1]), evaluate(p, D[2])]
-    Rnext = R[1] ∩ R[2]
+    dfX = df(X)
+    range_extrema, flag = _monotonicity_check(f, X, dfX)
+    flag && return hull(range_extrema, initial)
 
-    while  (Rprev.hi - Rnext.hi) >= tol * diam(Rnext) &&
-           (Rprev.lo - Rnext.lo) >= tol * diam(Rnext)
+    fX = f(X...)  # TODO: allow user to choose how to evaluate this (mean value, natural enclosure)
```

Comment:
We should open an issue to not forget...

Reply:
#77